test/scripts/github-action-helper.sh

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env bash
+
+set -xeEo pipefail
+
+#############
+# VARIABLES #
+#############
+: "${FUNCTION:=${1}}"
+
+# source https://github.com/rook/rook
+function find_extra_block_dev() {
+  # shellcheck disable=SC2005 # redirect doesn't work with sudo, so use echo
+  echo "$(sudo lsblk)" >/dev/stderr # print lsblk output to stderr for debugging in case of future errors
+  # relevant lsblk --pairs example: (MOUNTPOINT identifies boot partition)(PKNAME is Parent dev ID)
+  # NAME="sda15" SIZE="106M" TYPE="part" MOUNTPOINT="/boot/efi" PKNAME="sda"
+  # NAME="sdb"   SIZE="75G"  TYPE="disk" MOUNTPOINT=""          PKNAME=""
+  # NAME="sdb1"  SIZE="75G"  TYPE="part" MOUNTPOINT="/mnt"      PKNAME="sdb"
+  boot_dev="$(sudo lsblk --noheading --list --output MOUNTPOINT,PKNAME | grep boot | awk '{print $2}')"
+  echo "  == find_extra_block_dev(): boot_dev='$boot_dev'" >/dev/stderr # debug in case of future errors
+  # --nodeps ignores partitions
+  extra_dev="$(sudo lsblk --noheading --list --nodeps --output KNAME | grep -v loop | grep -v "$boot_dev" | head -1)"
+  echo "  == find_extra_block_dev(): extra_dev='$extra_dev'" >/dev/stderr # debug in case of future errors
+  echo "$extra_dev"                                                       # output of function
+}
+
+: "${BLOCK:=$(find_extra_block_dev)}"
+
+# source https://github.com/rook/rook
+use_local_disk() {
+  BLOCK_DATA_PART="/dev/${BLOCK}1"
+  sudo apt purge snapd -y
+  sudo dmsetup version || true
+  sudo swapoff --all --verbose
+  sudo umount /mnt
+  # search for the device since it keeps changing between sda and sdb
+  sudo wipefs --all --force "$BLOCK_DATA_PART"
+}
+
+deploy_rook() {
+  kubectl create -f https://raw.githubusercontent.com/rook/rook/master/deploy/examples/common.yaml
+  kubectl create -f https://raw.githubusercontent.com/rook/rook/master/deploy/examples/crds.yaml
+  kubectl create -f https://raw.githubusercontent.com/rook/rook/master/deploy/examples/operator.yaml
+  wait_for_operator_pod_to_be_ready_state
+  curl https://raw.githubusercontent.com/rook/rook/master/deploy/examples/cluster-test.yaml -o cluster-test.yaml
```

Comment:
disable the cephcsi driver deployed by Rook.

Reply:
done